src/main/kotlin/gg/essential/elementa/components/ScrollComponent.kt

```diff
@@ -378,17 +432,24 @@ class ScrollComponent @JvmOverloads constructor(
         needsUpdate = true
     }
 
-    private fun onScroll(delta: Float, isHorizontal: Boolean) {
-        if (isHorizontal) {
-            horizontalOffset += delta * pixelsPerScroll * currentScrollAcceleration
-        } else {
-            verticalOffset += delta * pixelsPerScroll * currentScrollAcceleration
+    /**
+     * @return whether the offset changed
+     */
+    private fun onScroll(delta: Float, isHorizontal: Boolean): Boolean {
+        var changed = false
+        val offset = if (isHorizontal) ::horizontalOffset else ::verticalOffset
+        val range = calculateOffsetRange(isHorizontal)
+        val newOffset = if(range.isEmpty()) innerPadding else offset.get() + delta * pixelsPerScroll * currentScrollAcceleration
+        if (newOffset in range) {
+            changed = true
+            offset.set(newOffset.coerceIn(range))
         }
```

Comment:
This looks like it would make it impossible to scroll all the way up/down unless you happen to exactly hit it?